drivers/scd30/scd30.c

```diff
@@ -0,0 +1,365 @@
+        return SCD30_NO_NEW_DATA;
+    }
+    ret = _scd30_read_data(dev, result);
+    if (ret != SCD30_OK) {
+        DEBUG("[scd30] Error reading values");
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int scd30_start_periodic_measurement(scd30_t *dev, uint16_t *interval,
+                                     uint16_t *apc)
+{
+    int ret = 0;
+
+    /* Check if input is valid */
+    if (!_is_valid_interval(interval)) {
+        DEBUG("[scd30] Interval value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+    if (!_is_valid_press_comp(apc)) {
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_set_param(dev, SCD30_INTERVAL, *interval);
+    if (ret != 0) {
+        return ret;
+    }
+    ret = scd30_set_param(dev, SCD30_START, *apc);
+    if (ret != 0) {
+        return ret;
+    }
+    return SCD30_OK;
+}
+
+int8_t scd30_stop_measurements(scd30_t *dev)
+{
+    uint8_t buffer[2];
+    uint16_t cmd = SCD30_STOP;
+    int ret;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    i2c_release(SCD30_I2C);
+    if (ret != 0) {
+        DEBUG("[scd30]: Error stopping measurements.\n");
+    }
+    return SCD30_OK;
+}
+
+/**
+ *      intern function to read data
+ *
+ *      @param  dev      device
+ *		@param  result	 struct in which data read is stored
+ *      @return          SCD30_OK on success
+ */
+static int8_t _scd30_read_data(scd30_t *dev, scd30_measurement_t *result)
+{
+    uint8_t buffer[18];
+    int ret = 0;
+    uint16_t cmd = SCD30_DATA;
+
+    buffer[0] = (uint8_t)(cmd >> 8);
+    buffer[1] = (uint8_t)cmd;
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_write_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 2, 0);
+    if (ret != 0) {
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    ret = i2c_read_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 18, 0);
+    if (ret != 0) {
+        DEBUG("[scd30] read_data: ret %i.\n", ret);
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    i2c_release(SCD30_I2C);
+
+    if (_scd30_crc_check(buffer, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 3, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 6, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 9, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 12, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+    if (_scd30_crc_check(buffer + 15, 2) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+
+    result->co2_concentration = _raw_val_to_uint32(byteorder_bebuftohl(&buffer[0]));
+    result->temperature = _raw_val_to_uint32(byteorder_bebuftohl(&buffer[6]));
+    result->relative_humidity = _raw_val_to_uint32(byteorder_bebuftohl(&buffer[12]));
```

Comment:
```suggestion
    result->co2_concentration = (float)byteorder_bebuftohl(&buffer[0]);
    result->temperature = (float)byteorder_bebuftohl(&buffer[6]);
    result->relative_humidity = (float)byteorder_bebuftohl(&buffer[12]);
```

Reply:
Cannot convert IEEE754 represented bits to actual float value like this. 
For it either we can do `*(float *)&val` (where val is the uint32_t variable with ieee754 value) but strict aliasing is broken because of this 
Therefore, this method of declaring union is required.

Reply:
Or we just keep that `_raw_val_to_float` helper function and drop the multiplication.